src/WasmDis.ts

```diff
@@ -1171,3 +1229,177 @@ export class NameSectionReader {
+            case ExternalKind.Function:
+              this._setName(this._functionNames, this._functionImportsCount++, importName, false);
+              break;
+            case ExternalKind.Table:
+              this._setName(this._tableNames, this._tableImportsCount++, importName, false);
+              break;
+            case ExternalKind.Memory:
+              this._setName(this._memoryNames, this._memoryImportsCount++, importName, false);
+              break;
+            case ExternalKind.Global:
+              this._setName(this._globalNames, this._globalImportsCount++, importName, false);
+              break;
+            default:
+              throw new Error(`Unsupported export ${importInfo.kind}`);
+          }
+          break;
+        case BinaryReaderState.NAME_SECTION_ENTRY:
+          var nameInfo = <INameEntry>reader.result;
+          if (nameInfo.type === NameType.Function) {
+            var functionNameInfo = <IFunctionNameEntry>nameInfo;
+            functionNameInfo.names.forEach((naming: INaming) => {
+              this._setName(this._functionNames, naming.index, bytesToString(naming.name), true);
+            });
+            this._hasNames = true;
+          } else if (nameInfo.type === NameType.Local) {
+            var localNameInfo = <ILocalNameEntry>nameInfo;
+            localNameInfo.funcs.forEach(localName => {
+              this._functionLocalNames[localName.index] = [];
+              localName.locals.forEach((naming: INaming) => {
+                this._functionLocalNames[localName.index][naming.index] = bytesToString(naming.name);
+              });
+            });
+            this._hasNames = true;
+          }
+          break;
+        case BinaryReaderState.EXPORT_SECTION_ENTRY:
+          var exportInfo = <IExportEntry>reader.result;
+          const exportName = this._generateExportName(exportInfo.field);
+          if (!exportName)
+            continue;
+          this._hasNames = true;
+          switch (exportInfo.kind) {
+            case ExternalKind.Function:
+              this._setName(this._functionNames, exportInfo.index, exportName, false);
+              break;
+            case ExternalKind.Table:
+              this._setName(this._tableNames, exportInfo.index, exportName, false);
+              break;
+            case ExternalKind.Memory:
+              this._setName(this._memoryNames, exportInfo.index, exportName, false);
+              break;
+            case ExternalKind.Global:
+              this._setName(this._globalNames, exportInfo.index, exportName, false);
+              break;
+            default:
+              throw new Error(`Unsupported export ${exportInfo.kind}`);
+          }
+          break;
+        default:
+          throw new Error(`Expectected state: ${reader.state}`);
+      }
+    }
+  }
+
+  public hasValidNames(): boolean {
+    return this._hasNames;
+  }
+
+  public getNameResolver(): INameResolver {
+    if (!this.hasValidNames())
```

Comment:
I think we don't need this check. This should always yield a `DevToolsNameResolver` even if no `names` section is found, and just fall back to the import/export name generation (or to generated names).

Reply:
Ok! Thanks, removed it!